### PR TITLE
Add required and suggested PHP extensions to composer.json

### PR DESCRIPTION
**Pull request by JoshuaLuckers**
### What does it do?
This adds required and suggested PHP extensions to composer.json.

### Why is it needed?
We want to check and suggest the system requirements when installing MODX via composer.

### Related issue(s)/PR(s)
Fixes issue #14048

### CODE REVIEW

**Review by Mark-H — DISMISSED** [review]

**Review by alroniks — DISMISSED** [review]

**Review by opengeek — CHANGES_REQUESTED** [review]
I don't think ext-imagick is required for MODX is it? Is this a new requirement? Because I've not installed this extension in quite some time and am now blocked from installing. :P

**Comment by alroniks** [comment]
Jason is right. Imagick can be replaced (and it does) by php-gd. It works better but it is not installed everywhere so we can not lock MODX to use this extension. I suggest moving it to recommended section @JoshuaLuckers

**Comment by JoshuaLuckers** [comment]
It’s listed as one of the [server requirements in the docs](
https://docs.modx.com/revolution/2.x/getting-started/server-requirement).

I will do the suggested changes later today. 👍

**Comment by alroniks** [comment]
But php-gd is really required for thumbnails and imagick is missed in recommended section.

**Comment by JoshuaLuckers** [comment]
> But php-gd is really required for thumbnails and imagick is missed in recommended section.

My misunderstanding!

**Comment by alroniks** [comment]
Not sure, but maybe it does not make sense to install imagic in travis.

**Review by alroniks — APPROVED** [review]

**Comment by alroniks** [comment]
Now it looks better. @opengeek

**Review by Mark-H — APPROVED** [review]
Not sure if this can be merged with a `changes requested` review from the other code owner.. but LGTM.

**Comment by alroniks** [comment]
@Mark-H We have to wait for approval from @opengeek in this case.

**Comment by JoshuaLuckers** [comment]
@opengeek I made the suggested changes.

**Review by opengeek — APPROVED** [review]

**Pull request merged**
